lighthouse-core/lib/page-functions.js

```diff
@@ -183,15 +194,10 @@ function getNodePath(node) {
   }
 
   const path = [];
-  while (node && node.parentNode) {
+  while (node && !!getNodeParent(node)) {
     const index = getNodeIndex(node);
     path.push([index, node.nodeName]);
-    if (node.parentNode.nodeType === Node.DOCUMENT_FRAGMENT_NODE) {
-      path.push([0, node.parentNode.nodeName]);
-      node = node.parentNode.host;
```

Comment:
JS style wise i think it's a little awkward that we got our while loop, but within this conditional we hop over an iteration that it was going to make.  

the `getNodeParent` method i extracted made the loop design a bit cleaner IMO

Reply:
yep i didn't really like this either since we weren't just naturally iterating through it. good idea moving the function out. overall it seems the part i had missed/got wrong was the index in the end (`a` vs `0`) but i had no clue where they were coming from (chrome i guess). if you ever find a way to do that test, that both implementations match up, that'd be incredibly useful.